src/mlmc/sampling_pool_pbs.py

```diff
@@ -286,23 +308,72 @@ def get_result_files(self, finished_pbs_jobs, unfinished_pbs_jobs):
                 # Split results to levels
                 for level_id, results in successful.items():
                     successful_results.setdefault(level_id, []).extend(results)
-                for level_id, results in failed_results.items():
+                for level_id, results in failed.items():
                     failed_results.setdefault(level_id, []).extend(results)
-                for level_id, results in times.items():
+                for level_id, results in time.items():
                     times[level_id] = results
 
                 # Delete pbsID file - it means job is finished
                 SamplingPoolPBS.delete_pbs_id_file(file)
 
-        # @TODO: get sample ids for failed pbs jobs and add these samples to failed ones with particular error message
+        if self._unfinished_sample_ids:
+            successful_results, failed_results, times = self._collect_unfinished(successful_results,
+                                                                                 failed_results, times)
 
         return successful_results, failed_results, n_running, times
-
-    def have_permanent_sample(self, sample_id):
+    
+    def _collect_unfinished(self, successful_results, failed_results, times):
         """
-        List of permanent samples or find per call?
+        Collect samples which had finished after main process crashed, append them to new collected samples
+        :param successful_results: dict
+        :param failed_results: dict
+        :param times: dict
+        :return: all input dictionaries
         """
+        already_collected = []
 
+        for sample_id in self._unfinished_sample_ids:
+            if sample_id in already_collected:
```

Comment:
Why? Can the samples appear multiple times in the unfinished list?
If necessary use a set instead of the list to prevent a square complexity.